tests/codex/sales/testsales.nim

```diff
@@ -296,8 +296,7 @@ asyncchecksuite "Sales":
                          slot: UInt256,
                          onBatch: BatchProc): Future[?!void] {.async.} =
       let blk = bt.Block.new( @[1.byte] ).get
-      discard await onBatch( blk.repeat(request.ask.slotSize.truncate(int)) )
-      return success()
+      await onBatch( blk.repeat(request.ask.slotSize.truncate(int)) )
```

Comment:
I think this returns a `Result` object, which needs to be handled.

Reply:
with `if =? ...` or `without res =? ...`

Reply:
Since this is a void result object, my thought is that I can just return it as it is from `onBatch` to the caller so they can see if there were errors, if any.

Reply:
Huh, I think this is still hitting the return error? `await onBatch` returns a `Result`, which isn't being handled and nim isn't picking that up?

Reply:
Wait, no, not in that case: `await onBatch` will evaluate to `Return[void, ref CatchableError]`, which also happens to be the return type for this async proc (`Future[?!void]`). So what happens is that in case of an error, that error will simply bubble up to the caller of `onBatch` via the result object.

Reply:
ah, you're right, it's a closure 👍